### PR TITLE
User authentication and registration implemented

### PR DESCRIPTION
**Pull request by SlobodanVeselinov**
- routes for registering and login a user are created
- controllers are created
- validation for register and login is set
- changed users table - added first name and last name columns
- added password confirmation

After registering user is automatically logged in and redirected to dashboard page.

- logout functionality is added and also remember me

the design of the pages should be next, and also creating the default layouts for the dashboard

### CODE REVIEW

**Comment by davorminchorov** [comment]
[Implement User Registration](https://trello.com/c/74XFE6em/8-implement-user-registration)

**Comment by davorminchorov** [comment]
[Implement User Authentication](https://trello.com/c/n6cCULLt/9-implement-user-authentication)

File: app/Http/Controllers/Auth/LoginController.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+namespace App\Http\Controllers\Auth;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Auth\LoginRequest;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+
+class LoginController extends Controller
+{
+    public function index()
```
Comment (davorminchorov) [inline app/Http/Controllers/Auth/LoginController.php]: Please add return types to all of the methods.

File: app/Http/Controllers/Auth/LoginController.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            'error' => 'The provided credentials do not match our records.',
+        ]);
+
+    }
+
+    public function logout()
+    {
+
+        Auth::logout();
+
```
Comment (davorminchorov) [inline app/Http/Controllers/Auth/LoginController.php:51]: Besides logging out, you are missing regeneration of the token and invalidating the old session.

File: app/Http/Requests/Auth/LoginRequest.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'email' => 'required|email',
```
Comment (davorminchorov) [inline app/Http/Requests/Auth/LoginRequest.php]: Please use the array syntax for all of the validation rules.

**Review by davorminchorov — DISMISSED** [review]
The code looks good, except for the few things that I pointed out in the comments.

Also, please use separate implementations for each ticket / branch, and avoid stacking multiple features in a single ticket for the future tickets. 

If you need to reuse the previous branch and it depends on the previous ticket, you can create a branch from the existing feature branch instead of main and point that out as a note in the pull request description.

The idea is to have all features isolated on a branch so that we can only revert changes by pieces and those changes won't depend on other features.

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 4]
```diff
@@ -12,7 +12,8 @@ public function up(): void
     {
         Schema::create('users', function (Blueprint $table) {
             $table->id();
-            $table->string('name');
+            $table->string('firstName');
+            $table->string('lastName');
```
Comment (davorminchorov) [inline database/migrations/2014_10_12_000000_create_users_table.php]: please use snake case for the column names to keep them consistent

File: app/Http/Controllers/Auth/LoginController.php (hunk 2 above)
Comment (SlobodanVeselinov) [reply app/Http/Controllers/Auth/LoginController.php:51]: fixed

File: app/Http/Requests/Auth/LoginRequest.php (hunk 3 above)
Comment (SlobodanVeselinov) [reply app/Http/Requests/Auth/LoginRequest.php]: fixed

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 4 above)
Comment (SlobodanVeselinov) [reply database/migrations/2014_10_12_000000_create_users_table.php]: fixed

File: app/Http/Controllers/Auth/LoginController.php (hunk 1 above)
Comment (SlobodanVeselinov) [reply app/Http/Controllers/Auth/LoginController.php]: I could use some help about returning types. I haven`t used return types declared to a method so far.

Comment (SlobodanVeselinov) [reply app/Http/Controllers/Auth/LoginController.php]: I have added RedirectResponse type to LoginController -> login method and RegisterController -> store method and it works fine. I can`t find what type should return when using Inertia::render to return inertia view

**Comment by davorminchorov** [comment]
Please resolve the merge conflicts.

File: app/Http/Controllers/Auth/LoginController.php (hunk 1 above)
Comment (davorminchorov) [reply app/Http/Controllers/Auth/LoginController.php]: Here's some info on type declarations: https://www.php.net/manual/en/language.types.declarations.php

Comment (davorminchorov) [reply app/Http/Controllers/Auth/LoginController.php]: You can check the return types for each Inertia method here: https://github.com/inertiajs/inertia-laravel/blob/master/src/Inertia.php

File: app/Http/Controllers/Auth/LoginController.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Auth\LoginRequest;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+
+class LoginController extends Controller
+{
+    public function index()
```
Comment (davorminchorov) [inline app/Http/Controllers/Auth/LoginController.php]: \Inertia\Response

Comment (SlobodanVeselinov) [reply app/Http/Controllers/Auth/LoginController.php]: resolved

File: app/Http/Controllers/MainPlatform/DashboardController.php [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\Http\Controllers\MainPlatform;
+
+use App\Http\Controllers\Controller;
+use Inertia\Inertia;
+
+class DashboardController extends Controller
+{
+    public function index()
```
Comment (davorminchorov) [inline app/Http/Controllers/MainPlatform/DashboardController.php]: Return type is missing here.

Comment (SlobodanVeselinov) [reply app/Http/Controllers/MainPlatform/DashboardController.php]: Added now

**Review by SlobodanVeselinov — COMMENTED** [review]
Nothing is styled, just to have links for those pages so you don`t have to maually type the route

**Review by davorminchorov — APPROVED** [review]

**Pull request merged**